termux-exec.c

```diff
@@ -56,12 +92,23 @@ int execve(const char* filename, char* const* argv, char *const envp[])
 
 	// execve(2): "A maximum line length of 127 characters is allowed
 	// for the first line in a #! executable shell script."
-	char shebang[128];
-	ssize_t read_bytes = read(fd, shebang, sizeof(shebang) - 1);
-	if (read_bytes < 5 || !(shebang[0] == '#' && shebang[1] == '!')) goto final;
+	char header[128];
+	ssize_t read_bytes = read(fd, header, sizeof(header) - 1);
+
+	// If we are executing a non-native ELF file, unset LD_PRELOAD.
+	// This avoids CANNOT LINK EXECUTABLE errors when running 32-bit code
+	// on 64-bit.
+	if (read_bytes >= 20 && !memcmp(header, "\x7f\x45\x4c\x46", 4)) {
```

Comment:
Maybe `#define` the header bytes earlier in the file? Or even use the definition in elf.h (ELFMAG0, ELFMAG1, ELFMAG2, ELFMAG3)

Reply:
Fixed